packages/e2e/test/wallet_epoch_3/PersonalWallet/delegation.test.ts

```diff
@@ -153,21 +156,30 @@ describe('PersonalWallet/delegation', () => {
     // Updates total and available balance after tx is on-chain
     expect(tx1ConfirmedState.balance.total.coins).toBe(expectedCoinsAfterTx1);
     expect(tx1ConfirmedState.balance.total).toEqual(tx1ConfirmedState.balance.available);
-    expect(tx1PendingState.balance.deposit).toEqual(stakeKeyDeposit);
+    expect(tx1ConfirmedState.balance.deposit).toEqual(stakeKeyDeposit);
 
     // If less than two epochs have elapsed, delegatee will still delegate to former pool during current epoch
     // if more than two epochs has elapsed, delegatee will delegate to new pool.
-    if (tx1ConfirmedState.epoch - initialState.epoch < 2) {
-      expect(tx1ConfirmedState.rewardAccount.delegatee?.currentEpoch?.id).toEqual(
-        initialState?.rewardAccount.delegatee?.currentEpoch?.id
-      );
-      expect(tx1ConfirmedState.rewardAccount.delegatee?.nextEpoch?.id).toEqual(
-        initialState?.rewardAccount.delegatee?.nextEpoch?.id
-      );
-      expect(tx1ConfirmedState.rewardAccount.delegatee?.nextNextEpoch?.id).toEqual(poolId);
-    } else {
-      expect(tx1ConfirmedState.rewardAccount.delegatee?.currentEpoch?.id).toEqual(poolId);
-    }
+    const numEpochsPassed = tx1ConfirmedState.epoch - initialState.epoch;
+    expect(tx1ConfirmedState.rewardAccount.delegatee?.currentEpoch?.id).toEqual(
+      numEpochsPassed === 0
+        ? initialState?.rewardAccount.delegatee?.currentEpoch?.id
+        : numEpochsPassed === 1
+        ? !initialState.isStakeKeyRegistered
```

Comment:
I'm not sure if this logic is correct. It might be if we take `initialState` snapshot *before* tx is in flight. Will keep this for now as it's passing, but will add info for follow-up ticket scope